scene/animation/animation_tree.h

```diff
@@ -131,6 +128,10 @@ class AnimationNode : public Resource {
 	GDVIRTUAL0RC(bool, _has_filter)
 
 public:
+	void blend_animation(const StringName &p_animation, double p_time, double p_delta, bool p_seeked, bool p_is_external_seeking, real_t p_blend, Animation::LoopedFlag p_looped_flag = Animation::LOOPED_FLAG_NONE);
+	double blend_node(const StringName &p_sub_path, Ref<AnimationNode> p_node, double p_time, bool p_seek, bool p_is_external_seeking, real_t p_blend, FilterAction p_filter = FILTER_IGNORE, bool p_sync = true, bool p_test_only = false);
+	double blend_input(int p_input, double p_time, bool p_seek, bool p_is_external_seeking, real_t p_blend, FilterAction p_filter = FILTER_IGNORE, bool p_sync = true, bool p_test_only = false);
+
```

Comment:
@fire Right. These are called multiple times in a single frame for the blending process and are not intended to be used by the user. They must be protected.

Reply:
These methods are already bound in class db and accessible to users: https://docs.godotengine.org/en/stable/classes/class_animationnode.html#class-animationnode-method-blend-animation

This visibility change only affects other engine code.

If they are only bound for technical reasons (tbh it doesn't look like it) and should never be called by users I can ofc removed the compatibility methods again, but the docs should be really clear about that as well.

Reply:
Probably the reason it is bound is to allow users to do blending in it when they implement custom AnimationNode. However, at least it should not be public.

Reply:
That seems outside of the scope of this PR. The change in this PR is only to enable the classless compatibility method, and it doesn't affect userland. If this API is implemented badly in principle, it should be fixed with a follow-up.

Reply:
Can you list the steps you wish us to take in sequence?

Reply:
1. Decide if this change somehow manages to break anything and thus cannot be part of this PR.
2. Decide if these methods being publicly accessible through ClassDB (as is currently the case) is an actual problem.
3. If answer 2 is yes, come up with a solution and create a PR implementing it.

Reply:
I believe we need to do this then @TokageItLab :

1. Animation Node methods being publicly accessible through ClassDB (as is currently the case) is an problem.
2. Come up with a solution and create a PR implementing it
3. This PR can then be rebased onto master. As far as I know this pr is required to convert the AnimationNode to public access.

Reply:
This PR here can be merged as is since it adds compatibility for some past changes. If you plan to un-expose some methods in AnimationNode, then we'd still need to have these compatibility methods, and then some more for the methods removed. That's kind of the point of having a compatibilty system.

Reply:
In any case, since the bound method needs to be removed later, it would be less trouble to remove the bound method once it is bound for the future.

Reply:
I'll just summarize this as misusing these methods breaks things and this change isn't as is isn't going to make anything worse.

Thus I'll mark this discussion as resolved and leave you to break compatibility in another PR.